### PR TITLE
[Perf] Implement perf summary and history for iOS performance test

### PR DESCRIPTION
**Pull request by taoran6**
**Linked GitHub issue ID**: #  

## Pull Request Checklist

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Code compiles correctly with all tests are passed.
- [ ] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [ ] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [ ] Yes
- [x] No

## Pull Request Description

**A few words to explain your changes**:  

### How you tested it
*Please make sure the change is tested, you can test it by adding UTs, do local test and share the screenshots, etc.*

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Technical design
- [ ] Build related changes
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe): 

### Feature UI screenshots or Technical design diagrams
*If this is a relatively large or complex change, kick it off by drawing the tech design with PlantUML and explaining why you chose the solution you did and what alternatives you considered, etc...*
image

### CODE REVIEW

**Review by dinghong2233 — DISMISSED** [review]

File: common/src/main/java/com/microsoft/hydralab/performance/entity/IOSMemoryPerfInfo.java [hunk 1]
```diff
@@ -24,12 +24,14 @@ public class IOSMemoryPerfInfo implements Serializable, IBaselineMetrics {
     @Override
     @JSONField(serialize = false)
     public LinkedHashMap<String, Double> getBaselineMetricsKeyValue() {
-        return null;
+        LinkedHashMap<String, Double> baselineMap = new LinkedHashMap<>();
+        baselineMap.put("memoryMB", (double) memoryMB);
+        return baselineMap;
```
[the remark this answers is not in the document]
Comment (taoran6) [reply common/src/main/java/com/microsoft/hydralab/performance/entity/IOSMemoryPerfInfo.java:29]: There will be no exception under this scenario. For more details, please refer to [PerformanceTestResultEntity::initBaselineMetrics()](https://github.com/microsoft/HydraLab/blob/main/common/src/main/java/com/microsoft/hydralab/common/entity/common/PerformanceTestResultEntity.java#L72)

**Review by dinghong2233 — DISMISSED** [review]

File: common/src/main/java/com/microsoft/hydralab/performance/parsers/IOSEnergyGaugeResultParser.java [hunk 2]
```diff
@@ -74,6 +74,8 @@ public PerformanceTestResult parse(PerformanceTestResult performanceTestResult)
                         lineNumber++;
                     }
                     performanceTestResult.performanceInspectionResults = newPerfInspectionResults;
+//                    performanceTestResult.setResultSummary(getSumIOSEnergy(newPerfInspectionResults));
```
Comment (AC-Phoenix) [inline common/src/main/java/com/microsoft/hydralab/performance/parsers/IOSEnergyGaugeResultParser.java:77]: This line of code is no longer needed?

File: common/src/main/java/com/microsoft/hydralab/performance/parsers/IOSEnergyGaugeResultParser.java [hunk 3]
```diff
@@ -82,4 +84,67 @@ public PerformanceTestResult parse(PerformanceTestResult performanceTestResult)
         }
         return performanceTestResult;
     }
+
+    private IOSEnergyGaugeInfo getAverageIOSEnergy(List<PerformanceInspectionResult> inspectionResults) {
+        if (inspectionResults == null || inspectionResults.size() == 0) {
+            return null;
+        }
+
+        IOSEnergyGaugeInfo sumEnergyInfo = new IOSEnergyGaugeInfo();
```
Comment (AC-Phoenix) [inline common/src/main/java/com/microsoft/hydralab/performance/parsers/IOSEnergyGaugeResultParser.java:93]: It seems that this variable holds the average and not the sum. How about `averageEnergyInfo`

Comment (taoran6) [reply common/src/main/java/com/microsoft/hydralab/performance/parsers/IOSEnergyGaugeResultParser.java]: Nice catch. I've revised

File: common/src/main/java/com/microsoft/hydralab/performance/parsers/IOSEnergyGaugeResultParser.java (hunk 2 above)
Comment (taoran6) [reply common/src/main/java/com/microsoft/hydralab/performance/parsers/IOSEnergyGaugeResultParser.java]: Syned with @dinghong2233, we may keep both summary types of `SUM` and `AVERAGE`. I added a comment to explain it.

**Review by AC-Phoenix — APPROVED** [review]

**Review by dinghong2233 — APPROVED** [review]

**Pull request merged**